generator/mapper.go

```diff
@@ -0,0 +1,123 @@
+package generator
+
+import (
+	"os"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/opencontrol/oscalkit/types/oscal/catalog"
+	"github.com/opencontrol/oscalkit/types/oscal/profile"
+)
+
+//CreateCatalogsFromProfile maps profile controls to multiple catalogs
+func CreateCatalogsFromProfile(profileArg *profile.Profile) ([]*catalog.Catalog, error) {
+
+	done := 0
+	errChan := make(chan error)
+	doneChan := make(chan bool)
+	var outputCatalogs []*catalog.Catalog
+	//Get first import of the profile (which is a catalog)
+	for _, profileImport := range profileArg.Imports {
+		go func(profileImport profile.Import) {
+			//ForEach Import's Href, Fetch the Catalog JSON file
+			catalogReference, err := GetCatalogFilePath(profileImport.Href.String())
+			if err != nil {
+				logrus.Errorf("invalid file path: %v", err)
+				doneChan <- true
+				return
+			}
+
+			f, err := os.Open(catalogReference)
+			if err != nil {
+				logrus.Errorf("cannot read file: %v", err)
+				doneChan <- true
+				return
+			}
+
+			//Once fetched, Read the catalog JSON and Marshall it to Go struct.
+			importedCatalog, err := ReadCatalog(f)
+			if err != nil {
+				logrus.Errorf("cannot parse catalog listed in import.href %v", err)
+				errChan <- err
+				return
+
+			}
+			//Prepare a new catalog object to merge into the final List of OutputCatalogs
+			newCatalog, err := getMappedCatalogControlsFromImport(importedCatalog, profileImport)
+			if err != nil {
+				errChan <- err
+				return
+			}
+			outputCatalogs = append(outputCatalogs, &newCatalog)
```

Comment:
Surround the slice with a mutex to prevent data race

```suggestion
                         mut.Lock()
 			outputCatalogs = append(outputCatalogs, &newCatalog)
 			mut.Unlock()
```